lib/tpm2_eventlog.c

```diff
@@ -0,0 +1,143 @@
+#include <inttypes.h>
+#include <stdlib.h>
+#include <tss2/tss2_tpm2_types.h>
+
+#include "log.h"
+#include "tcg-efi-event.h"
+#include "tpm2_eventlog.h"
+
+/* Functions to parse / process the structures that make up the eventlog. */
+size_t sizeof_alg(UINT16 alg_id) {
```

Comment:
This is a duplicate function of tpm2_alg_util_get_hash_size, and it's also missing SM3_256...

Reply:
Thx. I missed the `tpm2_alg_util` while porting this over.